### ISSUE TITLE
When VM is in execution, VBoxManage and VirtualBox manager hang up

### ISSUE THREAD

**Issue by mgallizio-dlabs** [onlyfang/VBoxVmService#52]
I'm using VBoxVmManage and the service is working great, however I noticed that when a VM is in execution using a service user account, either VirtualBox user interface and VBoxManage become slow and unresponsive. This happens also with a toy service I made in .net that starts a VM via VBoxManage.exe and VBoxHeadless.exe.  

This is what happens: I start the service, my VM boots up and works fine. then I open the VirtualBox frontend. Usually it takes about 1~2 seconds to start, but now it requires up to 30 seconds. Then it seems to work fine for a minute or two, but then it start to hang up again.

The same thing happens using VBoxManage. Usually when I start it I get a response on stdout in about 1 second for simple commands (i.e. listing virtual machines with "vboxmanage.exe list vms"), but now it requires up to 30 seconds to get its work done.  

If I stop VBoxVmService (or the headless VM), everything works well again.

I tried everything I could, using different services (homemade and other similar projects I found on the Internet), using different VirtualBox versions, enabling logs using the VBOXSVC_RELEASE_LOG* system env variables but I found nothing that could help me. Also a Google search for the problem was inconclusive. 

I don't thing this problem is related to my Windows installation, because on another PC I own I get the same problems. 

I'm using the latest release of Windows 10 Pro (20H2), patched with all the available updates.

I know that probably this problem is not related to VBoxVmService itself, but may I ask if someone else experiences similar issues? If so, is there a known workaround to get VirtualBox work fine even if a headless VM is running as service?

Thanks in advance!

**Comment by onlyfang** [comment]
There are known issue to run [VirtualBox GUI together with VBoxVmService](https://github.com/onlyfang/VBoxVmService/wiki/Runing-VirtualBox-GUI-together-with-VBoxVmService). I don't know why but suspect the performance issue might also be related to that.

**Comment by mgallizio-dlabs** [comment]
Yes, I was aware or the issues specified in the link you gave me, in my particular case they don't represent a problem... but the slowdown could be...
So there is no known solution to resolve this issue? (I always think it's a VirtualBox problem, not a VBoxVmService one, but I may be wrong)

Thank you

**Comment by slam7E5** [comment]
I saw the this slowdown with Windows 10 Professional and 2019 Server installations. Moreover, a virtual Machine (VM) started via the VBoxVMService is not accessible via Oracle's GUI. No terminal window, no stopping via Machine-Close. This is not a problem for me, as
1) SSH allows accessing into the VM.
2) for a running VM, virtualisation settings cannot be changed before stopping the VM anyway. The taskbar of VBoxVMService enables stopping the VM. Then, Oracle GUI is normally usable as I only run one virtualised appliance and its stopping voids the slowness issue.
3) while Oracle GUI does not show that the VM is running, it does not enable double starting, but spits out an error message. This is good, since the service I use must not be run in two concurrent instances to avoid given conflicts.

**Comment by mark-chk** [comment]
Usually is not a problem for me too, the VMs managed by VBoxVmService are working fine, but I have some cases where I need to use the virtual machine as service and one or two VMs that I run directly from VirtualBox (and for these I require the GUI to work with them). And I can't simply stop the service VM because is always used by other users (via RDP)...
